indexService/main.js

```diff
@@ -31,126 +36,124 @@ var findH = function(array) {
       if (i !== -1) {
-        results[i].count++;
-      //else it is not yet in the array
+        results[i].count += 1;
+      // else it is not yet in the array
       } else {
         author.count = 1;
         results.push(author);
       }
     });
   });
 
-  // console.log('Pre-filter: ', results.map(result => result.id), results.map(result => result.count));
-  //filter out authors who have less than 5 posts on the subject
+  // filter out authors who have less than 5 posts on the subject
   results = results.filter(result => result.count > 4);
-  // console.log('Post-filter: ', results.map(result => result.id));
-  //sort the results by highest h-index first 
+  // sort the results by highest h-index first
   results.sort((a, b) => b.hIndex - a.hIndex);
-  // console.log('Post-sorting: ', results.map(result => result.id), results.map(result => result.hIndex));
-  //return a mapped array of just their ids
+  // return a mapped array of just their ids
   return results.map(result => result.id);
 };
 
-//attrs is an object of attributes and the ranking functions used to calculate their values
-//options is used when fetching posts from tags in order to include authors for author ranking
-var rank = function(Model, attrs, cb, options) {
-
-  var start = new Date();
-
-  Model.findAll().then(results => {
-
-    return Promise.all(results.map(instance => {
-      return instance.getPosts(options).then(posts => {
-
-        //FOR PETE - CHANGE THIS TO SORT ON b.rank - a.rank ONCE PAGERANK IS IMPLEMENTED
-        posts.sort((a, b) => b.inLinks.length - a.inLinks.length);
-
-        var obj = {};
-        for (var key in attrs) {
-          obj[key] = attrs[key](posts);
-        }
-
-        return instance.updateAttributes(obj);
-      });
-    }));
-
-  }).then(updated => {
-
-    var end = new Date();
-    cb(null, updated, end - start);
-
-  }).catch(err => {
-
-    cb(err);
+// Page Rank
+const pageRank = (cb) => {
+  let posts;
+  Post.max('postId')
+    .then(n =>
+      Post.findAndCountAll({ order: ['postId'] })
+        .then((results) => {
+          // pass results down
+          posts = results.rows;
+          return solver.makeAdjacencyMatrix(results.rows, n);
+        })
+        // M and v are matrices, see mathjs docs
+        .then(M => solver.solver(M, 0.8, 0.001))
+        .catch(err => cb(err, null)))
+    .then(v =>
+      Promise.all(posts.map(post =>
+        post.updateAttributes({ rank: v.get([post.postId - 1, 0]) })
+      ))
+    )
+    .then(rankedPosts => cb(null, rankedPosts))
+    .catch(err => cb(err, null));
+};
 
-  });
 
+// attrs is an object of attributes and the ranking functions used to calculate their values
+// options is used when fetching posts from tags in order to include authors for author ranking
+const rank = (Model, attrs, cb, options) => {
+  const start = new Date();
+
+  Model.findAll()
+    .then(results =>
+      Promise.all(results.map(instance =>
```

Comment:
Did these returns that were there become implicit? Did you check that these still work without explicit returns?